server.js

```diff
@@ -182,23 +183,36 @@ app.use(function (req, res, next) {
 app.locals.gtmId = gtmId
 app.locals.asset_path = '/public/'
 app.locals.useAutoStoreData = (useAutoStoreData === 'true')
+app.locals.useCookieSessionStore = (useCookieSessionStore === 'true')
 app.locals.cookieText = config.cookieText
 app.locals.promoMode = promoMode
 app.locals.releaseVersion = 'v' + releaseVersion
 app.locals.serviceName = config.serviceName
 
-// Support session data
-app.use(session({
+// Session uses service name to avoid clashes with other prototypes
+const sessionName = 'govuk-prototype-kit-' + (Buffer.from(config.serviceName, 'utf8')).toString('hex')
+let sessionOptions = {
+  secret: sessionName,
```

Comment:
not sure if there's an advantage to using a predictable string for the 'secret' of the cookie? I think it should be somewhat secret?

Reply:
ah, is this because when the app restarts it won't know the randomly generated secret? Hmm.. maybe we could store the secret somehow. If needed

Reply:
Yeah that's the trade-off sadly.

The old approach re-hashed the secret on every restart. This would lock you out of your cookie causing it to be regenerated.

Sounds like an idea though.